### PR TITLE
fix(frontend): Comments from the newly opened comment section are displayed in the previously opened comment sections tm-417

### PR DESCRIPTION
**Pull request by tarmiel**
demo-comments

### CODE REVIEW

**Review by Vfivew — APPROVED** [review]

File: apps/frontend/src/modules/activities/slices/activities.slice.ts [hunk 1]
```diff
@@ -15,22 +14,18 @@ import {
...
 const initialState: State = {
 	activities: [],
-	activityComments: [],
-	createCommentDataStatus: DataStatus.IDLE,
 	dataStatus: DataStatus.IDLE,
-	getActivityCommentsDataStatus: DataStatus.IDLE,
 	likeDataStatus: DataStatus.IDLE,
 };
 
+const ZERO_INDEX = 0;
```
Comment (what1s1ove) [inline apps/frontend/src/modules/activities/slices/activities.slice.ts:27]: B28

File: apps/frontend/src/pages/feed/libs/components/activity-comments/activity-comments.tsx [hunk 2]
```diff
@@ -19,31 +21,39 @@ type Properties = {
...
-		useAppSelector((state) => {
-			return {
-				comments: state.activities.activityComments,
-				isLoadingComments:
-					state.activities.getActivityCommentsDataStatus === DataStatus.PENDING,
-				isLoadingCreateComment:
-					state.activities.createCommentDataStatus === DataStatus.PENDING,
-			};
-		});
+	const [comments, setComments] = useState<CommentWithRelationsResponseDto[]>(
```
Comment (what1s1ove) [inline apps/frontend/src/pages/feed/libs/components/activity-comments/activity-comments.tsx:24]:
```suggestion
```
you do not need to keep it in the state. you should use redux store instead

File: apps/frontend/src/pages/feed/libs/components/activity-comments/activity-comments.tsx [hunk 3]
```diff
@@ -19,31 +21,39 @@ type Properties = {
...
+		[],
+	);
+	const [isLoadingComments, setIsLoadingComments] = useState<boolean>(false);
+	const [isLoadingCreateComment, setIsLoadingCreateComment] =
```
Comment (what1s1ove) [inline apps/frontend/src/pages/feed/libs/components/activity-comments/activity-comments.tsx:28]:
```suggestion
```
use data status from redux store

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: apps/frontend/src/pages/feed/libs/components/activity-comments/activity-comments.tsx (hunk 2 above)
Comment (tarmiel) [reply apps/frontend/src/pages/feed/libs/components/activity-comments/activity-comments.tsx:24]: I need each comment section to be indepenedent, so when I open comments under one activity it should not effect others.

File: apps/frontend/src/pages/feed/libs/components/activity-comments/activity-comments.tsx (hunk 3 above)
Comment (tarmiel) [reply apps/frontend/src/pages/feed/libs/components/activity-comments/activity-comments.tsx:28]: We do not want to share loading state between opened comment sections

Comment (what1s1ove) [reply apps/frontend/src/pages/feed/libs/components/activity-comments/activity-comments.tsx:28]: got it, then do not use a boolean for loading. use a data status instead

File: apps/frontend/src/pages/feed/libs/components/activity-comments/activity-comments.tsx [hunk 4]
```diff
@@ -19,31 +21,39 @@ type Properties = {
...
 
 	useEffect(() => {
-		void dispatch(activityActions.getAllCommentsToActivity(activityId));
+		setIsLoadingComments(true);
+		void dispatch(activityActions.getAllCommentsToActivity(activityId))
+			.unwrap()
+			.then((response) => {
+				setComments(response.items);
+				setIsLoadingComments(false);
+			});
```
Comment (what1s1ove) [inline apps/frontend/src/pages/feed/libs/components/activity-comments/activity-comments.tsx:56]:
```suggestion
```
you can not take it this way. take it from the store using the `useSelector`

File: apps/frontend/src/pages/feed/libs/components/activity-comments/activity-comments.tsx (hunk 2 above)
Comment (what1s1ove) [reply apps/frontend/src/pages/feed/libs/components/activity-comments/activity-comments.tsx:24]: you can not do it this way - 
```ts
			.then((response) => {
				setComments(response.items);
				setIsLoadingComments(false);
			});
```

**Review by what1s1ove — CHANGES_REQUESTED** [review]

**Review by YourTeaChan — APPROVED** [review]

File: apps/frontend/src/pages/feed/libs/components/activity-comments/activity-comments.tsx (hunk 2 above)
Comment (what1s1ove) [reply apps/frontend/src/pages/feed/libs/components/activity-comments/activity-comments.tsx:26]: it is not resolved

File: apps/frontend/src/pages/feed/libs/components/activity-comments/activity-comments.tsx [hunk 5]
```diff
@@ -19,33 +23,47 @@ type Properties = {
...
-		[activityId, dispatch],
+		[activityId, comments, dispatch],
 	);
 
 	useEffect(() => {
-		void dispatch(activityActions.getAllCommentsToActivity(activityId));
+		setLoadCommentsDataStatus(DataStatus.PENDING);
+		void dispatch(activityActions.getAllCommentsToActivity(activityId))
+			.unwrap()
+			.then((response) => {
```
Comment (what1s1ove) [inline apps/frontend/src/pages/feed/libs/components/activity-comments/activity-comments.tsx:58]: is not resolved

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: apps/frontend/src/pages/feed/libs/components/activity-comments/activity-comments.tsx [hunk 6]
```diff
@@ -52,13 +50,13 @@ const ActivityComments: React.FC<Properties> = ({ activityId }: Properties) => {
 				<Loader color="orange" size="large" />
 			) : (
 				<div className={styles["comments-container"]}>
-					{comments.map((comment) => (
+					{comments?.map((comment) => (
```
Comment (what1s1ove) [inline apps/frontend/src/pages/feed/libs/components/activity-comments/activity-comments.tsx:53]: at this stage you should be sure that there are comments

**Review by what1s1ove — CHANGES_REQUESTED** [review]

**Review by what1s1ove — APPROVED** [review]

**Pull request merged**